File: web/src/core/email/EmailGenerator.ts
import type { ReviewRow } from '../../types/domain';

export interface GeneratedEmail {
  subject: string;
  body: string;
  mailtoHref: string;
}

function formatDate(date: Date): string {
  return date.toLocaleDateString('en-US', { year: 'numeric', month: 'long', day: 'numeric' });
}

export function generateEmail(rows: ReviewRow[], date: Date = new Date()): GeneratedEmail {
  const subject = `Daily Production Status - ${formatDate(date)}`;

  const projectBlocks = rows.map((row) =>
    [`Project ${row.projectNumber.value}`, `Customer: ${row.customer.value}`, `Estimated Days Remaining: ${row.daysRemaining.value}`].join(
      '\n',
    ),
  );

  const body = ['Daily Production Status', '', 'Project Updates', '', projectBlocks.join('\n\n'), '', 'End of Report'].join('\n');

  const mailtoHref = `mailto:?subject=${encodeURIComponent(subject)}&body=${encodeURIComponent(body)}`;

  return { subject, body, mailtoHref };
}

File: app/src/main/java/com/productionboard/scanner/report/EmailGenerator.kt
package com.productionboard.scanner.report

import com.productionboard.scanner.domain.ReviewRow
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

data class GeneratedEmail(val subject: String, val body: String)

/**
 * Builds the daily status email subject/body from the rows the user has
 * approved (checked Include). Never sends anything - EmailScreen only
 * offers copy-to-clipboard and Android's share sheet (ACTION_SEND),
 * which hands the text to whatever app the user picks (Gmail, Outlook,
 * Teams, ...).
 */
object EmailGenerator {

    fun generate(rows: List<ReviewRow>, date: Date = Date()): GeneratedEmail {
        val included = rows.filter { it.included }
        val dateStr = SimpleDateFormat("MMMM d, yyyy", Locale.US).format(date)
        val subject = "Daily Production Status - $dateStr"

        val projectBlocks = included.map { row ->
            listOf(
                "Project ${row.projectNumber.value}",
                "Customer: ${row.customer.value}",
                "Estimated Days Remaining: ${row.daysRemaining.value}",
            ).joinToString("\n")
        }

        val body = listOf(
            "Daily Production Status",
            "",
            "Project Updates",
            "",
            projectBlocks.joinToString("\n\n"),
            "",
            "End of Report",
        ).joinToString("\n")

        return GeneratedEmail(subject, body)
    }
}
